### PR TITLE
Fix in combat idle logout notification and add config option for notification

### PR DESCRIPTION
**Pull request by MagicfTail**
The idle logout timer is 20 min instead of 5 min when in combat

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 1]
```diff
@@ -318,6 +352,8 @@ public void onGameTick(GameTick event)
 		}
 	}
 
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: remove these 2 new lines

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 2]
```diff
@@ -275,19 +281,47 @@ public void onGameStateChanged(GameStateChanged gameStateChanged)
 		}
 	}
 
+	@Subscribe
+	public void onHitsplat(HitsplatApplied event)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: onHitsplatApplied

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 3]
```diff
@@ -275,19 +281,47 @@ public void onGameStateChanged(GameStateChanged gameStateChanged)
...
+	{
+		final Hitsplat hitsplat = event.getHitsplat();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: this should be after the if as it serves no point before if you return early from the method

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 4]
```diff
@@ -275,19 +281,47 @@ public void onGameStateChanged(GameStateChanged gameStateChanged)
...
+			return;
+		}
+
+		final Hitsplat hitsplat = event.getHitsplat();
+
+		if (hitsplat.getHitsplatType() == Hitsplat.HitsplatType.DAMAGE
+			|| hitsplat.getHitsplatType() == Hitsplat.HitsplatType.BLOCK)
+		{
+			inCombat = true;
+			countdown = HIGHEST_MONSTER_ATTACK_SPEED;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: You should reset both inCombat and countdown in resetTimers

**Comment by deathbeam** [comment]
ALso you should probably add unit tests for this (now that idle notifier has some)

**Comment by francomidence** [comment]
you say this was fixed but it isn't @deathbeam 
Any setting I should change or something?

**Comment by Nightfirecat** [comment]
You should join our Discord server and ask in the `#support` channel. Given that this PR was merged 33 years ago, it's more likely a config issue than a problem with the plugin.

**Pull request merged**